src/hooks/useUrlState.tsx

```diff
@@ -0,0 +1,24 @@
+import { useSearchParams } from 'react-router-dom';
+import type { Timer } from '../types/timers';
+
+export const useUrlState = () => {
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const getTimersFromUrl = () => {
+        const timersParam = searchParams.get('timers');
+        if (!timersParam) return [];
+        try {
+            return JSON.parse(decodeURIComponent(timersParam));
+        } catch {
+            return [];
+        }
+    };
+
+    const setTimersInUrl = (timers: Timer[]) => {
+        const encodedTimers = encodeURIComponent(JSON.stringify(timers));
```

Comment:
long encoded url's are kind of ugly- you could consider encoding them in base64, and then decoding when needed. It's also ugly, but a little less so in my opinion :)